#Ansible Module Name -- rds_acc_mgmt.py
#Test Case : Grant Account permissions to db instance that does not belong to specified RDS instance
#Ansible Playbook created by - Abha H
---
- name: Grant Account permissions to db instance that does not belong to specified RDS instance
  hosts: localhost
  connection: local
  vars_files:
     - ../../../vars/alicloud-creds.yml
     - ../../vars/test-environment.yml
     - ../../vars/general-parameters.yml
   
  tasks:
   - block :
     - name: Scenario 1 - Grant Account permissions to db instance that does not belong to specified RDS instance
       rds_acc_mgmt:
         acs_access_key: '{{ aliyun_acs_access_key }}'
         acs_secret_access_key: '{{ aliyun_acs_secret_access_key }}'
         region: cn-hongkong
         command: '{{ commandGrant }}'
         db_instance_id: '{{ instanceIdDiff }}'
         account_name: demo
         db_name: '{{ dbName }}' 
         account_privilege: '{{ accountPriveledgeGrant }}'
       register: result_scn1
     - debug: var=result_scn1   

     rescue:
      - fail:
          msg: "The fail block is added just to handle exception. This block won't be executed ever."
        when: no



     always:
      - assert:  
         that:
            - "'failed' in result_scn1"
#            - "'The specified account is not found.' in result_scn1.msg[0]['Error Message']"
         msg: 'Test pass FAILED !!!'     